R/util.R

```diff
@@ -1,9 +1,10 @@
 #' @importFrom stringr str_match
 #' @keywords internal
 parse_url = function(url) {
-    parsed = as.list(stringr::str_match(url, "(.*)://(.*):(\\d*)"))
-    parsed = parsed[-1]
-    names(parsed) <- c("provider", "host", "port")
+    #parsed = as.list(stringr::str_match(url, "(.*)://(.*):(\\d*)"))
+    parsed = as.list(stringr::str_match(url, "(.*)://(:(.*)@)?([^:/]*)(:(\\d*))?(/(\\d*))?"))
```

Comment:
Maybe we can make some small modifications to this regex to make it work with different brokers and url patterns. 

Generally, brokers/databases urls comes in the format of `engine://user:password@host/db` where user and password are optional. So we'd basically need to add a new capture group for user and replace the pattern matching for the db part so it matches not only to digits, making it compatible with other engines besides Redis.

What do you think? 😃

Reply:
I've updated the regex to capture any username that is provided. As a confirmation I tested `parse_url` on a variety of URL forms:

url | provider | username | password | host | port | db
-- | -- | -- | -- | -- | -- | --
redis://redis1/1 | redis | NA | NA | redis1 | NA | 1
redis://redis1:6379/1 | redis | NA | NA | redis1 | 6379 | 1
redis://redis1:6379 | redis | NA | NA | redis1 | 6379 | NA
redis://:mytestpassword@redis1:6379/1 | redis | NA | mytestpassword | redis1 | 6379 | 1
redis://:mytestpassword@redis1:6379 | redis | NA | mytestpassword | redis1 | 6379 | NA
redis://:mytestpassword@redis1 | redis | NA | mytestpassword | redis1 | NA | NA
redis://:mytestpassword@redis1/1 | redis | NA | mytestpassword | redis1 | NA | 1
redis://username:mytestpassword@redis1 | redis | username | mytestpassword | redis1 | NA | NA
redis://username:mytestpassword@redis1/1 | redis | username | mytestpassword | redis1 | NA | 1
redis://username:mytestpassword@redis1/abc | redis | username | mytestpassword | redis1 | NA | abc
redis://username:mytestpassword@redis1:6379 | redis | username | mytestpassword | redis1 | 6379 | NA
redis://username:mytestpassword@redis1:6379/abc | redis | username | mytestpassword | redis1 | 6379 | abc

I also adapted `RedisQueue` and `RedisBackend` to accept `username` as parameter and ignore it (so that `queue` and `backend` don't have to determine appropriate parameters per queue type). As a note, `redux::hiredis` doesn't actually fail if the username is passed as a parameter but it does show a warning so I left it out for now. It would be easy enough to add in for future compatibility in case usernames are accepted in the future.

Reply:
That's amazing! This will make it a lot easier for future integrations :)